pkg/apis/toolchain/v1alpha1/toolchainstatus_types.go

```diff
@@ -27,9 +27,12 @@ const (
 	ToolchainStatusRegServiceResourceNotFoundReason = "RegServiceResourceNotFound"
 
 	// member status reasons
-	ToolchainStatusMemberStatusNoClustersFoundReason   = "NoMemberClustersFound"
-	ToolchainStatusMemberStatusNotFoundReason          = "MemberStatusNotFound"
-	ToolchainStatusMemberToolchainClusterRemovedReason = "MemberToolchainClusterRemoved"
+	ToolchainStatusMemberStatusNoClustersFoundReason         = "NoMemberClustersFound"
+	ToolchainStatusMemberStatusNotFoundReason                = "MemberStatusNotFound"
+	ToolchainStatusMemberToolchainClusterRemovedReason       = "MemberToolchainClusterRemoved"
+	ToolchainStatusMemberStatusConsoleRouteUnavailableReason = "ConsoleRouteUnavailable"
+	ToolchainStatusMemberStatusCheRouteUnavailableReason     = "CheRouteUnavailable"
```

Comment:
By setting this reason we define Che as a hard dependency or necessary condition for Dev Sandbox platform. I mean, if I wanted to deploy the operators to some custom clusters where Che is not available, then I won't have a way to make it running - The resulting `MemberStatus` and thus also `ToolchainStatus` won't have `AllComponentsReady` status.

Reply:
From my point of view, if the Che is not installed (and thus URL is not available), then the `CheDashboardURL` param should be empty and - as a consequnce - the host operator should not set anything it in the MUR status.
I guess that the current implementation does something similar.

Reply:
I'm not sure about Che dependency either.
It would be great to have something for Che routes monitoring but I agree that it should be rather optional.
I would still keep it here as a reason. It's just a const. How it's used is another story. And it's up to the controller to fail the entire status with that reason or use it differently.

Reply:
I understand your point - just don't forget to take care of it in the other PR here: https://github.com/codeready-toolchain/member-operator/pull/208/files#diff-46039918795b512c2c3c08d32a609f0143cea872da1a2caef3592b8334c5c23eR377-R380 and don't set the error when the route is not found.
Would you mind renaming the constant/reason then? For example
```suggestion
	ToolchainStatusMemberStatusUnableToGetCheRouteReason     = "UnableToGetCheRoute"
```
This message more shows that the problem is with getting, not with the existence.

Reply:
Well.. I was thinking of using this reason when/if we want to report any problem with accessing the che dashboard route. Without specifics.
Whether we want to set the entire ready status condition to false with this reason if there is a problem accessing the Che route in the controller or not is a separate question.
There are pros and cons. Maybe we should keep it optional.. Let's say by default our configuration doesn't require Che to be available but we can make it required in our configuration so the controller starts using the reason and fail the entire ready condition if so.

Reply:
@MatousJobanek I'm merging this PR but let's keep discussing it. This constant does not affect CRDs anyway so it's pretty easy to change it anytime.